staff/sergi-pallice/course-2/task2/html.js

```diff
@@ -1,28 +1,183 @@
-const request = new XMLHttpRequest();
-request.open('GET', requestURL);
-request.responseType = 'json';
-request.send();
-
-request.onload = function() {
-    const superHeroes = request.response;
-    console.log(superHeroes);
-    document.getElementById("js").innerHTML = superHeroes.squadName;
-    // populateHeader(superHeroes);
-    // showHeroes(superHeroes);
-  }
\ No newline at end of file
+const states = {
+  "AL": "Alabama",
+  "AK": "Alaska",
+  "AS": "American Samoa",
+  "AZ": "Arizona",
+  "AR": "Arkansas",
+  "CA": "California",
+  "CO": "Colorado",
+  "CT": "Connecticut",
+  "DE": "Delaware",
+  "DC": "District Of Columbia",
+  "FM": "Federated States Of Micronesia",
+  "FL": "Florida",
+  "GA": "Georgia",
+  "GU": "Guam",
+  "HI": "Hawaii",
+  "ID": "Idaho",
+  "IL": "Illinois",
+  "IN": "Indiana",
+  "IA": "Iowa",
+  "KS": "Kansas",
+  "KY": "Kentucky",
+  "LA": "Louisiana",
+  "ME": "Maine",
+  "MH": "Marshall Islands",
+  "MD": "Maryland",
+  "MA": "Massachusetts",
+  "MI": "Michigan",
+  "MN": "Minnesota",
+  "MS": "Mississippi",
+  "MO": "Missouri",
+  "MT": "Montana",
+  "NE": "Nebraska",
+  "NV": "Nevada",
+  "NH": "New Hampshire",
+  "NJ": "New Jersey",
+  "NM": "New Mexico",
+  "NY": "New York",
+  "NC": "North Carolina",
+  "ND": "North Dakota",
+  "MP": "Northern Mariana Islands",
+  "OH": "Ohio",
+  "OK": "Oklahoma",
+  "OR": "Oregon",
+  "PW": "Palau",
+  "PA": "Pennsylvania",
+  "PR": "Puerto Rico",
+  "RI": "Rhode Island",
+  "SC": "South Carolina",
+  "SD": "South Dakota",
+  "TN": "Tennessee",
+  "TX": "Texas",
+  "UT": "Utah",
+  "VT": "Vermont",
+  "VI": "Virgin Islands",
+  "VA": "Virginia",
+  "WA": "Washington",
+  "WV": "West Virginia",
+  "WI": "Wisconsin",
+  "WY": "Wyoming"
+}
+
+/*------------------------------------------USER VALUES---------------------------------------- */
+document.getElementById("democrat").addEventListener("click", function () { filter_party() });
+document.getElementById("republican").addEventListener("click", function () { filter_party() });  
+document.getElementById("independent").addEventListener("click", function () { filter_party() });
+document.getElementById("state_filter").addEventListener("input",  function () { filter_party() });
+
+/*---------------------------------------AUTO-CREATE TABLE---------------------------------------- */
+function makeMemberRows(members_array){
+  document.getElementById("senate-data").innerHTML = "";  // Deletes previous data in div
+  let table = document.getElementById("senate-data");     // We'll put the table inside div "senate-data"
+  let thead = document.createElement('thead');            // Variable to create block header
+  let tbody = document.createElement("tbody");            // Variable to create body
+  /*-------------------------------------------TABLE HEADER---------------------------------------- */
+  let tr      = document.createElement('tr');
```

Comment:
Maybe a loop to do all repetitive lines of code since this one (line 77) until line 130.